x-pack/plugins/security_solution/server/lib/entity_analytics/risk_engine/get_user_risk_engine_privileges.ts

```diff
@@ -7,52 +7,27 @@
 
 import type { KibanaRequest } from '@kbn/core/server';
 import type { SecurityPluginStart } from '@kbn/security-plugin/server';
-import type { RiskEnginePrivilegesResponse } from '../types';
 import {
   RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES,
   RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES,
 } from '../../../../common/risk_engine';
+import { checkAndFormatPrivileges } from '../utils/check_and_format_privileges';
 
-const groupPrivilegesByName = <PrivilegeName extends string>(
-  privileges: Array<{
-    privilege: PrivilegeName;
-    authorized: boolean;
-  }>
-): Record<PrivilegeName, boolean> => {
-  return privileges.reduce<Record<string, boolean>>((acc, { privilege, authorized }) => {
-    acc[privilege] = authorized;
-    return acc;
-  }, {});
-};
+const mutableClusterPrivileges = [...RISK_ENGINE_REQUIRED_ES_CLUSTER_PRIVILEGES];
+const mutableIndexPrivileges = JSON.parse(JSON.stringify(RISK_ENGINE_REQUIRED_ES_INDEX_PRIVILEGES));
```

Comment:
I am looking for some typescript help here. Maybe I could pair with someone? 

We have the index privileges and cluster privileges which are objects with known keys, e.g https://github.com/elastic/kibana/pull/172441/files#diff-377522cacbb11f9aa89fbb1833c8bfc0aa13ac297d4c2f95fbd29adf8585df24R30 these are generated by code generation so cannot be meaningfully changed.

We then have privileges stored as constants, [here](https://github.com/elastic/kibana/pull/172441/files#diff-f87b466afb41f93c21bd13e2d01ff5bbb0e034bdb2a107bfe1b875e120242f1aR11) where the array can only contain items which match the keys above. 

I need to lookup privileges here https://github.com/elastic/kibana/blob/7d6bb4f6ec59c28b6e64535f25b1f0f5f1ca30cb/x-pack/plugins/security_solution/public/entity_analytics/components/risk_engine_privileges_callout/use_missing_risk_engine_privileges.ts#L73